ztests/suite/zqd/rec-curl.yaml

```diff
@@ -0,0 +1,84 @@
+script: |
+  source recserv.sh
+  echo ===
+  curl --header "Content-Type: application/json" -request POST \
+    --data '{"addr":"a.b.c:5000","node":"a.b"}' \
+    http://$ZQD_HOST/register 2> err
+  echo ===
+  curl --header "Content-Type: application/json" -request POST \
+    --data '{"N":1}' \
+    http://$ZQD_HOST/recruit 2> err
+  #
+  # For this second register, the worker will be in the reserved pool,
+  # so it will not be reregistered. This still returns a status of OK.
+  #
+  echo ===
+  curl --header "Content-Type: application/json" -request POST \
+    --data '{"addr":"a.b.c:5000","node":"a.b"}' \
+    http://$ZQD_HOST/register 2> err
+  #
+  # Now when recruit is called again, the reserved worker will not be recruited
+  #
+  echo ===
+  curl --header "Content-Type: application/json" -request POST \
+    --data '{"N":1}' \
+    http://$ZQD_HOST/recruit 2> err
+  #
+  # Now unreserve the worker so it can be reregistered
+  #
+  echo ===
+  curl --header "Content-Type: application/json" -request POST \
+    --data '{"addr":"a.b.c:5000"}' \
+    http://$ZQD_HOST/unreserve 2> err
+  echo ===
+  curl --header "Content-Type: application/json" -request POST \
+    --data '{"addr":"a.b.c:5000","node":"a.b"}' \
+    http://$ZQD_HOST/register 2> err
+  echo ===
+  curl --header "Content-Type: application/json" -request POST \
+    --data '{"N":1}' \
+    http://$ZQD_HOST/recruit 2> err
+  #
+  # Now register a node, then deregister it, and verify that
+  # it is not available to be recruited
+  #
```

Comment:
I for one find these story based, multiple cases in one tests hard to isolate when I'm unfamiliar with the code base but need to fix a breakage. I would prefer this test case be broken down into smaller isolated test cases.